examples/pytorch/evolveGCN/model.py

```diff
@@ -0,0 +1,201 @@
+        # default parameters follow the official config
+        super(EvolveGCNO, self).__init__()
+        self.num_layers = num_layers
+        self.recurrent_layers = nn.ModuleList()
+        self.gnn_convs = nn.ModuleList()
+        self.gcn_weights_list = nn.ParameterList()
+
+        # In the paper, EvolveGCN-O use LSTM as RNN layer. According to the official code,
+        # EvolveGCN-O use GRU as RNN layer. Here we follow the official code.
+        # See: https://github.com/IBM/EvolveGCN/blob/90869062bbc98d56935e3d92e1d9b1b4c25be593/egcn_o.py#L53
+        # PS: I try to use torch.nn.LSTM directly,
+        #     like [pyg_temporal](github.com/benedekrozemberczki/pytorch_geometric_temporal/blob/master/torch_geometric_temporal/nn/recurrent/evolvegcno.py)
+        #     but the performance is worse than use torch.nn.GRU.
+        # PPS: I think torch.nn.GRU can't match the manually implemented GRU cell in the official code,
+        #      we follow the official code here.
+        self.recurrent_layers.append(MatGRUCell(in_feats=in_feats, out_feats=n_hidden))
+        self.gcn_weights_list.append(Parameter(torch.Tensor(in_feats, n_hidden)))
+        self.gnn_convs.append(
+            GraphConv(in_feats=in_feats, out_feats=n_hidden, bias=False, activation=nn.RReLU(), weight=False))
+        for _ in range(num_layers - 1):
+            self.recurrent_layers.append(MatGRUCell(in_feats=n_hidden, out_feats=n_hidden))
+            self.gcn_weights_list.append(Parameter(torch.Tensor(n_hidden, n_hidden)))
+            self.gnn_convs.append(
+                GraphConv(in_feats=n_hidden, out_feats=n_hidden, bias=False, activation=nn.RReLU(), weight=False))
+
+        self.mlp = nn.Sequential(nn.Linear(n_hidden, classifier_hidden),
+                                 nn.ReLU(),
+                                 nn.Linear(classifier_hidden, n_classes))
+        self.reset_params()
+
+    def reset_params(self):
+        for layer in self.gcn_weights_list:
+            torch.nn.init.xavier_normal_(layer)
+
+    def forward(self, g_list):
+        feature_list = []
+        for g in g_list:
+            feature_list.append(g.ndata['feat'])
+        for i in range(self.num_layers):
+            W = self.gcn_weights_list[i]
+            for j, g in enumerate(g_list):
+                # Attention: I try to use the below code to set gcn.weight(similar to pyG_temporal),
+                # but it doesn't work. It seems that the gradient function lost in this situation,
+                # more discussion see here: https://github.com/benedekrozemberczki/pytorch_geometric_temporal/issues/80
+                # ====================================================
+                # W = self.gnn_convs[i].weight[None, :, :]
+                # W, _ = self.recurrent_layers[i](W)
+                # self.gnn_convs[i].weight = nn.Parameter(W.squeeze())
+                # ====================================================
```

Comment:
It won't work because the weight parameter is replaced as a whole.

Usually the preferred way to implement NN modules with intermediate results as weights is to express the NN module itself as a function (e.g. `torch.nn.Conv2d` is implemented by `torch.nn.functional.conv2d`, and similar for `torch.nn.functional.PReLU`).  For our GCN in particular though you can supply the GCN weight as `weight` keyword argument in `GraphConv.forward`.  This does not work for other modules.

@jermainewang what's your opinion on this case?  Should we put functionizing the NN modules into backlog?